lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EnsembleCalibration.py

```diff
@@ -246,15 +278,15 @@ def test_temperature_realizations_data_check(self):
             )
         else:
             predictor_data = np.array(
-                [[230.53659896, 241.80363361, 253.07066826],
-                 [264.33770292, 275.60473757, 286.87177222],
-                 [298.13880687, 309.40584153, 320.67287618]],
+                [[231.46936, 242.74925, 254.02914],
+                 [265.30902, 276.5889, 287.8688],
+                 [299.14868, 310.42856, 321.70844]],
                 dtype=np.float32
             )
             variance_data = np.array(
-                [[18.04589231, 18.04589231, 18.04589231],
-                 [18.04589231, 18.04589231, 18.04589231],
-                 [18.04589231, 18.04589231, 18.04589231]],
+                [[0.89248854, 0.89248854, 0.89248854],
```

Comment:
The variance has changed a lot! Is this expected?

Reply:
Running the master code, again without statsmodel to access these tests, with 1000 iterations gives the following variances:
```
2.392e-10, 2.392e-10, 2.392e-10, 2.392e-10, 2.392e-10, 2.392e-10, 2.392e-10, 2.392e-10, 2.392e-10
```
which are tiny and not like either of the expected sets. Each increase in the number of iterations leads to a decrease in the variances as can be seen below, suggesting even the new expected variances given here are not the converged results, rather the values should be closer to 0:
```
No. iterations : variance
200 : 18.04589231
300 : 17.2682
400 : 16.7999
500 : 0.7757
600 : 0.0595
800 : 2.392e-10
```

If I run this PR version of this test with 1000 iterations I get variances of 7.9191e-13.

Removing the warning handling we get the expected warning about failure to converge:
```
The final iteration resulted in a percentage change that is greater than the accepted threshold of 5% i.e. [ 2.46329139 10.18034888  0.41063477  6.30490672  0.26428811  0.54476245]. 
A satisfactory minimisation has not been achieved. 
Last iteration: [ 0.92869924 -0.05670054 -0.00925947  0.16589068  0.74932268  0.6431581 ], 
Last-but-one iteration: [ 0.90637264 -0.05146157 -0.00922161  0.17705375  0.74734753  0.64668098]
Absolute difference: [0.0223266  0.00523897 0.00003787 0.01116307 0.00197515 0.00352288]
```

I guess the question is what do we want these tests to do? They show a consistent approach towards the solution with 200 iterations, but they do not converge. Is this sufficient, or do we want a converged solution? 800 iterations gives a converged solution.